modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/ConvertProcessor.java

```diff
@@ -42,7 +42,11 @@
             @Override
             public Object convert(Object value) {
                 try {
-                    return Integer.parseInt(value.toString());
+                    String str = value.toString();
+                    if (str.startsWith("0x")) {
+                        return Integer.decode(str);
```

Comment:
should we add a test for when it is `Integer.decode` that throws the NumberFormatException?

same goes for `Long.decode`